apiserver/storage/storage.go

```diff
@@ -256,12 +256,17 @@ func filterAttachment(machineSet set.Strings, attachment volume.Attachment) (par
 	}
 
 	size := attachment.Size()
+
+	storageTag := names.StorageTagKind + "-unknown/0"
+	if names.IsValidStorage(attachment.Storage()) {
```

Comment:
IMO, it makes more sense to just check != "" here. The ID came from state, so it had better be valid or things will be broken. In the case of a non-empty, but invalid storage ID, panicking is more useful since it doesn't hide the problem. Alternatively use the ParseTag method and log an error.